src/App.tsx

```diff
@@ -54,95 +38,80 @@ const SidebarContainer = React.lazy(() => import('./containers/Sidebar').then(({
+};
 
-        if (isMobileDevice) {
-            return (
-                <div className="pg-mobile-app">
-                    <MobileHeader />
-                    <MobileFooter />
-                </div>
-            );
-        }
+const RenderDeviceContainers = () => {
+    const isMobileDevice = useSelector(selectMobileDeviceState);
 
+    if (isMobileDevice) {
         return (
-            <React.Fragment>
-                <HeaderContainer/>
-                <SidebarContainer/>
-                <CustomizationContainer/>
-                <AlertsContainer/>
+            <div className="pg-mobile-app">
+                <MobileHeader />
                 <LayoutContainer/>
-                <FooterContainer/>
-            </React.Fragment>
+                <MobileFooter />
+            </div>
         );
     }
 
-    public render() {
-        const { lang } = this.props;
-
-        return (
-            <IntlProvider locale={lang} messages={this.getTranslations()} key={lang}>
-                <Router history={history}>
-                    <ErrorWrapper>
-                        <React.Suspense fallback={null}>
-                            {this.renderDeviceContainers()}
-                        </React.Suspense>
-                    </ErrorWrapper>
-                </Router>
-            </IntlProvider>
-        );
-    }
-
-    private handleChangeWindowWidth = () => {
-        const { isMobileDevice } = this.props;
-        const height = window.innerHeight;
-        const width = window.innerWidth;
-        let updatedMobileDeviceState = false;
-
-        if ((width && width < 768) || (height && height < 600)) {
-            updatedMobileDeviceState = true;
-        }
-
-        if (isMobileDevice !== updatedMobileDeviceState) {
-            this.props.setMobileDevice(updatedMobileDeviceState);
-        }
-    };
-
-    private getTranslations = () => {
-        const { lang, isMobileDevice } = this.props;
-
-        if (isMobileDevice) {
-            return  {
-                ...languageMap[lang],
-                ...mobileTranslations[lang],
-            };
-        }
-
-        return languageMap[lang];
-    };
-}
-
-const mapStateToProps = (state: RootState): ReduxProps => ({
-    lang: selectCurrentLanguage(state),
-    isMobileDevice: selectMobileDeviceState(state),
-});
-
-const mapDispatchToProps: MapDispatchToPropsFunction<DispatchProps, {}> = dispatch => ({
-    setMobileDevice: payload => dispatch(setMobileDevice(payload)),
-});
+    return (
+        <React.Fragment>
+            <HeaderContainer/>
+            <SidebarContainer/>
+            <CustomizationContainer/>
+            <AlertsContainer/>
+            <LayoutContainer/>
+            <FooterContainer/>
+        </React.Fragment>
+    );
+};
+
+export const App = () => {
+    const dispatch = useDispatch();
+    const lang = useSelector(selectCurrentLanguage);
+    const isMobileDevice = useSelector(selectMobileDeviceState);
+
+    React.useEffect(() => {
```

Comment:
Do you want it to be called every render?

Reply:
I think it should be called once (like componentDidMount ?)

Reply:
good point, changed